tests/integration/services/activation/engine/test_ports.py

```diff
@@ -98,6 +108,63 @@ def test_ports_with_empty_plugin():
   sources:
     - ansible.eda.webhook:
 """
-    ports = find_ports(rulebook)
+    context = {"webhook_port": 8000}
+    ports = find_ports(rulebook, context)
 
     assert ports == []
+
+
+def test_ports_with_extra_vars():
+    rulebook = """
+---
+- name: Run a webhook listener service
+  hosts: all
+  sources:
+    - ansible.eda.webhook:
+        port: "{{ webhook_port | default(5000) }}"
+"""
+    context = {"webhook_port": 8000}
+    ports = find_ports(rulebook, context)
+    assert ports == [(None, 8000)]
```

Comment:
@hsong-rh Is the first element of the tuple a host or port, can we have a test with the filled value

Reply:
@mkanoor it's the host, let me add tests for it